spec/webauthn/public_key_credential_with_assertion_spec.rb

```diff
@@ -233,5 +235,111 @@
         end
       end
     end
+
+    context "when verifying a migrated U2F credential" do
+      let!(:credential) do
+        stored_credential = seeds[:u2f_migration][:stored_credential]
+        migrated_credential = WebAuthn::U2fMigrator.new(**stored_credential)
+
+        [
+          migrated_credential.credential.id,
+          migrated_credential.credential.public_key,
+          migrated_credential.authenticator_data.sign_count
+        ]
+      end
+
+      let(:public_key_credential) do
+        WebAuthn::PublicKeyCredentialWithAssertion.new(
+          type: credential_type,
+          id: credential_id,
+          raw_id: credential_raw_id,
+          client_extension_outputs: { appid: true },
+          response: assertion_response
+        )
+      end
+
+      let(:assertion_response) do
+        assertion_response = seeds[:u2f_migration][:assertion][:response]
+
+        WebAuthn::AuthenticatorAssertionResponse.new(
+          **assertion_response.transform_values { |v| Base64.strict_decode64(v) }
+        )
+      end
+
+      let(:origin) { seeds[:u2f_migration][:assertion][:origin] }
+      let(:challenge) { seeds[:u2f_migration][:assertion][:challenge] }
+
+      context "and appid is set in configuration file" do
+        let(:legacy_u2f_appid) { seeds[:u2f_migration][:stored_credential][:app_id] }
+
+        before do
+          WebAuthn.configuration.legacy_u2f_appid = legacy_u2f_appid
+        end
+
+        it "works" do
+          expect(
+            public_key_credential.verify(
+              challenge,
+              public_key: credential_public_key,
+              sign_count: credential_sign_count
+            )
+          ).to be_truthy
+        end
+
+        context "if appid extension is not requested" do
+          let(:public_key_credential) do
+            WebAuthn::PublicKeyCredentialWithAssertion.new(
+              type: credential_type,
+              id: credential_id,
+              raw_id: credential_raw_id,
+              response: assertion_response
+            )
+          end
+
+          it "fails" do
+            expect do
+              public_key_credential.verify(
+                challenge,
+                public_key: credential_public_key,
+                sign_count: credential_sign_count
+              )
+            end.to raise_error(WebAuthn::RpIdVerificationError)
+          end
+        end
+      end
+
+      context "and appid is not set in configuration file" do
+        it "raises an error" do
+          expect do
+            public_key_credential.verify(
+              challenge,
+              public_key: credential_public_key,
+              sign_count: credential_sign_count
+            )
+          end.to raise_error("Unspecified legacy U2F AppID")
```

Comment:
This is a tricky one tbh...

```
PublicKeyCredentialWithAssertion #verify when verifying a migrated U2F credential and appid is not set in configuration file raises an error
```

We currently support that you can call  `AuthenticatorAssertionResponse.from_get(extensions: { appid: { 'my_appid' }` – without the need of setting the appid in the config – and that will correctly send the extension. But doing it this way will raise an error if you use `PublicKeyCredentialWithAssertion#verify` – in the case that the extension output returning from the client was `true. 

In the U2F migration docs it states that you have to call `AuthenticatorAssertionResponse#verify` instead and pass the `appid` as the `rp_id`, but I feel that we could also accept the `rp_id` param in `PublicKeyCredentialWithAssertion#verify` so we don't force users to use the former – which is not that "smart" and does not perform all the verifications.